bindings/matrix-sdk-ffi/src/client_builder.rs

```diff
@@ -468,19 +478,28 @@ impl ClientBuilder {
         for certificate in builder.additional_root_certificates {
             // We don't really know what type of certificate we may get here, so let's try
             // first one type, then the other.
-            if let Ok(cert) = Certificate::from_der(&certificate) {
-                certificates.push(cert);
-            } else {
-                let cert =
-                    Certificate::from_pem(&certificate).map_err(|e| ClientBuildError::Generic {
-                        message: format!("Failed to add a root certificate {e:?}"),
+            match Certificate::from_der(&certificate) {
+                Ok(cert) => {
+                    certificates.push(cert);
+                }
+                Err(e) => {
+                    tracing::warn!("Failed loading certificate as DER, falling back to PEM: {e:?}");
```

Comment:
Ah maybe I should reduce this to an `info`? I added it because the DER certificate I was providing would fail to be load and I had no idea why as that error got swallowed up. Or I could include it in that `format!` so both failures are shown, that might be nicer.

Reply:
(The crate doesn't support PEM for iOS so the error returned was along the lines of "Not implemented on iOS" which wasn't helpful)

Reply:
I've included both messages in the error now 👍

Reply:
Ah that sounds annoying, indeed including both errors sounds like a good approach.